### PR TITLE
Query cleanup

### PR DESCRIPTION
**Pull request by radex**

### CODE REVIEW

File: src/QueryDescription/test.js [hunk 1]
```diff
@@ -410,6 +387,113 @@ describe('buildQueryDescription', () => {
       ]),
     )
   })
+  it('supports sorting query', () => {
```
Comment (radex) [inline src/QueryDescription/test.js:390]: just reordering

File: src/QueryDescription/test.js [hunk 2]
```diff
@@ -550,115 +634,3 @@ describe('queryWithoutDeleted', () => {
     )
   })
 })
-
```
Comment (radex) [inline src/QueryDescription/test.js:553]: again, just reordering, nothing to see here

File: src/adapters/lokijs/worker/encodeQuery/test.js [hunk 3]
```diff
@@ -27,21 +27,18 @@ const mockCollection = Object.freeze({
...
 })
 
-const testQuery = query => encodeQueryRaw(query.serialize())
 const encoded = clauses => encodeQueryRaw(new Query(mockCollection, clauses).serialize())
 
 describe('LokiJS encodeQuery', () => {
   it('encodes simple queries', () => {
-    const query = new Query(mockCollection, [])
-    expect(testQuery(query)).toEqual({
+    expect(encoded([])).toEqual({
```
Comment (radex) [inline src/adapters/lokijs/worker/encodeQuery/test.js:34]: no real changes in this file, just use encoded to dry

File: src/adapters/sqlite/encodeQuery/test.js [hunk 4]
```diff
@@ -28,24 +28,24 @@ const mockCollection = Object.freeze({
...
 })
 
-const encoded = clauses => encodeQuery(new Query(mockCollection, clauses))
+const encoded = (clauses, countMode) => encodeQuery(new Query(mockCollection, clauses), countMode)
 
 describe('SQLite encodeQuery', () => {
   it('encodes simple queries', () => {
-    const query = new Query(mockCollection, [])
-    expect(encodeQuery(query)).toBe(
+    expect(encoded([])).toBe(
```
Comment (radex) [inline src/adapters/sqlite/encodeQuery/test.js:35]: same thing here

**Comment by radex** [comment]
i wanted to do some quick cleanup and I knew it would make for a big diff even though there's almost no real changes here, so I made it a separate PR so you can just skim through it and not waste your time looking at a diff that contains no meaningful changes. There's only a few (minor) changes at the beginning and end

**Comment by radex** [comment]
‼️ don't merge - dependent branch

File: src/observation/encodeMatcher/test.js [hunk 5]
```diff
@@ -43,16 +56,21 @@ describe('SQLite encodeMatcher', () => {
...
-    const error = `can't be encoded into a matcher`
-    expect(() => makeMatcher([Q.on('projects', 'team_id', 'abcdef')])).toThrow(error)
-    expect(() => makeMatcher([Q.experimentalJoinTables(['foo'])])).toThrow(error)
-    expect(() => makeMatcher([Q.experimentalNestedJoin('foo', 'bar')])).toThrow(error)
-    expect(() => makeMatcher([Q.experimentalSortBy('left_column', 'asc')])).toThrow(error)
-    expect(() => makeMatcher([Q.experimentalTake(100)])).toThrow(error)
-    expect(() => makeMatcher([Q.experimentalTake(100)])).toThrow(error)
-    expect(() => makeMatcher([Q.unsafeLokiFilter(() => {})])).toThrow(error)
+    unencodableQueries.forEach(query => {
+      // console.log(query)
```
Comment (kokusGr) [inline src/observation/encodeMatcher/test.js:60]: Do you want to keep this? :)

File: src/observation/encodeMatcher/index.js [hunk 6]
```diff
@@ -59,17 +60,10 @@ const encodeConditions: (Where[]) => Matcher<*> = pipe(
...
-
   invariant(
-    !joinTables.length &&
-      !nestedJoinTables.length &&
-      !sortBy.length &&
-      !take &&
-      !skip &&
-      !lokiFilter,
+    canEncodeMatcher(query),
     `Queries with joins, sortBy, take, or skip can't be encoded into a matcher`,
```
Comment (kokusGr) [inline src/observation/encodeMatcher/index.js:65]: Hmm now it's going to be even harder to keep this comment up to date. Maybe add `forbiddenClauses` array to `canEncode.js` and here `join` into string?

File: src/observation/subscribeToSimpleQuery/index.js [hunk 7]
```diff
@@ -54,8 +54,6 @@ export default function subscribeToSimpleQuery<Record: Model>(
   // observeQueryWithColumns
   alwaysEmit: boolean = false,
 ): Unsubscribe {
-  invariant(!query.hasJoins, 'subscribeToSimpleQuery only supports simple queries!')
```
Comment (kokusGr) [inline src/observation/subscribeToSimpleQuery/index.js:57]: Hmm do we even need `hasJoins` now? I guess it is a part of Query API (although marked as internal) so not sure if we can remove it...

Comment (radex) [reply src/observation/subscribeToSimpleQuery/index.js:57]: Good point, I'll search to see if anything even uses this internally.

File: src/observation/encodeMatcher/index.js (hunk 6 above)
Comment (radex) [reply src/observation/encodeMatcher/index.js:65]: good point, i'll do that

File: src/observation/encodeMatcher/test.js (hunk 5 above)
Comment (radex) [reply src/observation/encodeMatcher/test.js:60]: I do! if you're running an expect() in a loop and one of those expectations fails, it's usually difficult to figure out which one is it - so i'll then add a clg to check which one failed. So i'm keeping this for DX :)

**Comment by radex** [comment]
@kokusGr Thank you again for your really swift CRs :)

File: src/observation/encodeMatcher/test.js (hunk 5 above)
Comment (kokusGr) [reply src/observation/encodeMatcher/test.js:60]: Make sense, just wanted to be sure :)

File: src/observation/subscribeToSimpleQuery/index.js (hunk 7 above)
Comment (radex) [reply src/observation/subscribeToSimpleQuery/index.js:57]: deprecated in next PR

File: src/observation/encodeMatcher/index.js (hunk 6 above)
Comment (radex) [reply src/observation/encodeMatcher/index.js:65]: fixed in next PR

**Pull request merged**